### PR TITLE
Add option to specify multiple sources in file

### PR DESCRIPTION
**Pull request by roshandathathri**

### CODE REVIEW

**Comment by roshandathathri** [comment]
@insertinterestingnamehere Any idea why some of the CI tests are failing?

**Comment by insertinterestingnamehere** [comment]
@Breakinbad can you take a look at the Travis CI failures? It looks like a recent change in bipart broke all our clang builds. Bipart also needs to have clang-format run on it again.

The Alpine LInux and CentOS builds in CircleCI appear to be caused by upstream changes in packages. I'll take a look at those some time in the next few days.

None of the errors appear to be related to this PR. Please be sure to run clang-format before merging though since that check is currently broken.

**Comment by Breakinbad** [comment]
Oh yes sorry. I had to make a change for the artifact eval of ppopp...

**Comment by insertinterestingnamehere** [comment]
No worries. Thanks for looking into it.

**Comment by Breakinbad** [comment]
Hi Ian,

Seems fine to me! I ran clang-format and it looks ok.

-Sepideh

**Comment by insertinterestingnamehere** [comment]
WRT clang-format, after loading the usual modules, run (in the main Galois directory) `bash scripts/check_format.sh -fix lonestar`, then commit the result.

**Comment by insertinterestingnamehere** [comment]
To see the failing clang builds, go to the main repo page and scroll down. Right below the big word "Overview" there are two status icons that link to the failing (or succeding) builds on CircleCI and Travis CI. The bad clang builds on Travis CI are at https://travis-ci.org/github/IntelligentSoftwareSystems/Galois if you want the direct link, though I need to migrate to their new ".com" service soon, so that link won't be stable long-term. There are some other issues with some of the CircleCI builds that I need to fix, but the clang-related failures are all present in the Travis builds. The unused variable looks easy to fix. I'm not sure about the OSX build errors, though those also look bipart-related.

**Comment by insertinterestingnamehere** [comment]
To build with clang locally, after loading the usual modules, configure with `cmake -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ .."

File: lonestar/analytics/cpu/betweennesscentrality/BetweennessCentrality.cpp [hunk 1]
```diff
@@ -38,35 +38,17 @@ namespace cll = llvm::cl;
...
-                   cll::desc("Level: Use for single source BC (default off)"),
-                   cll::init(false));
-
-static cll::opt<uint64_t>
-    startSource("startNode",
-                cll::desc("Level/Outer: Starting source node used for "
-                          "betweeness-centrality (default 0); works with "
-                          "singleSource flag only"),
-                cll::init(0));
+static cll::opt<std::string> startNodesFile(
```
Comment (gurbinder533) [inline lonestar/analytics/cpu/betweennesscentrality/BetweennessCentrality.cpp:41]: Correct me if am wrong but it seems it is reading sources from a file and then using all of them to run BC. For Intel experiments, we need another parameter that specifies how many sources to run bc application each time. So if that number is 4 and total sources are 64 read from a file, I would do 4 at a time and run bc 16 times.

**Review by gurbinder533 — APPROVED** [review]
LGTM. Thank you so much.

File: lonestar/analytics/cpu/bfs/bfs.cpp [hunk 2]
```diff
@@ -44,10 +44,17 @@ static const char* url = "breadth_first_search";
...
 static cll::opt<std::string>
     inputFile(cll::Positional, cll::desc("<input file>"), cll::Required);
-static cll::opt<unsigned int>
-    startNode("startNode",
-              cll::desc("Node to start search from (default value 0)"),
-              cll::init(0));
+static cll::opt<std::string> startNodesFile(
+    "startNodesFile",
+    cll::desc("File containing whitespace separated list of source "
+              "nodes for computing betweenness centrality; "
```
Comment (gurbinder533) [inline lonestar/analytics/cpu/bfs/bfs.cpp:50]: For computing bfs.

**Comment by l-hoang** [comment]
@Breakinbad It seems your changes are preventing this change (and #365) from getting merged. May you check the failing CI tests and get them fixed?

**Pull request closed without merging**